mm/2s2h/Enhancements/ResolutionEditor/ResolutionEditor.cpp

```diff
@@ -0,0 +1,503 @@
+        - IntegerScale.NeverExceedBounds                - Prevents manual resizing from exceeding screen bounds.
+
+    The following CVars are also implemented in LUS for niche use cases:
+        - IgnoreAspectCorrection                        - Stretch framebuffer to fill screen.
+            This is something of a power-user setting for niche setups that most people won't need or care about,
+            but may be useful if playing the Switch/Wii U ports on a 4:3 television.
+        - IntegerScale.ExceedBoundsBy                   - Offset the max screen bounds, usually by +1.
+                                                          This isn't that useful at the moment, so it's unused here.
+*/
+
+namespace AdvancedResolutionSettings {
+enum setting { UPDATE_aspectRatioX, UPDATE_aspectRatioY, UPDATE_verticalPixelCount };
+
+const char* aspectRatioPresetLabels[] = {
+    "Off", "Custom", "Original (4:3)", "Widescreen (16:9)", "Nintendo 3DS (5:3)", "16:10 (8:5)", "Ultrawide (21:9)"
+};
+const float aspectRatioPresetsX[] = { 0.0f, 16.0f, 4.0f, 16.0f, 5.0f, 16.0f, 21.0f };
+const float aspectRatioPresetsY[] = { 0.0f, 9.0f, 3.0f, 9.0f, 3.0f, 10.0f, 9.0f };
+const int default_aspectRatio = 1; // Default combo list option
+
+const char* pixelCountPresetLabels[] = { "Custom",     "Native N64 (240p)", "2x (480p)",       "3x (720p)", "4x (960p)",
+                                         "5x (1200p)", "6x (1440p)",        "Full HD (1080p)", "4K (2160p)" };
+const int pixelCountPresets[] = { 480, 240, 480, 720, 960, 1200, 1440, 1080, 2160 };
+const int default_pixelCount = 0; // Default combo list option
+
+// Resolution clamp values as hardcoded in Ship::Gui::ApplyResolutionChanges()
+const uint32_t minVerticalPixelCount = SCREEN_HEIGHT;
+const uint32_t maxVerticalPixelCount = 4320; // 18x native, or 8K TV resolution
+
+const unsigned short default_maxIntegerScaleFactor = 6; // Default size of Integer scale factor slider.
+
+enum messageType { MESSAGE_ERROR, MESSAGE_WARNING, MESSAGE_QUESTION, MESSAGE_INFO, MESSAGE_GRAY_75 };
+const ImVec4 messageColor[]{
+    { 0.85f, 0.0f, 0.0f, 1.0f },  // MESSAGE_ERROR
+    { 0.85f, 0.85f, 0.0f, 1.0f }, // MESSAGE_WARNING
+    { 0.0f, 0.85f, 0.85f, 1.0f }, // MESSAGE_QUESTION
+    { 0.0f, 0.85f, 0.55f, 1.0f }, // MESSAGE_INFO
+    { 0.75f, 0.75f, 0.75f, 1.0f } // MESSAGE_GRAY_75
+};
+
+void AdvancedResolutionSettingsWindow::InitElement() {
+}
+
+void AdvancedResolutionSettingsWindow::DrawElement() {
+    ImGui::SetNextWindowSize(ImVec2(497, 599), ImGuiCond_FirstUseEver);
+    if (ImGui::Begin("Advanced Resolution Settings", &mIsVisible)) {
+        // Initialise update flags.
+        bool update[3];
+        for (uint8_t i = 0; i < sizeof(update); i++)
+            update[i] = false;
+
+        // Initialise integer scale bounds.
+        short max_integerScaleFactor = default_maxIntegerScaleFactor; // default value, which may or may not get
```

Comment:
```suggestion
        int16_t max_integerScaleFactor = default_maxIntegerScaleFactor; // default value, which may or may not get
```
Just to keep it consistent, sticking to the size defined types is preferred if it's necessary to use a short here